modules/rostests/apitests/iphlpapi/GetInterfaceName.c

```diff
@@ -0,0 +1,459 @@
+/*
+ * PROJECT:         ReactOS API Tests
+ * LICENSE:         LGPLv2.1+ - See COPYING.LIB in the top level directory
+ * PURPOSE:         Tests for network interface name resolving functions
+ * PROGRAMMERS:     Stanislav Motylkov
+ */
+
+#include <apitest.h>
+
+#define WIN32_NO_STATUS
+#include <iphlpapi.h>
+#include <ndk/rtlfuncs.h>
+#include <rpc.h>
+
+static DWORD (WINAPI * pTestFunction)(LPVOID, LPVOID, PULONG, DWORD, DWORD);
+
+/*
+ * Tests for NhGetInterfaceNameFromGuid
+ */
+static
+VOID
+test_NhGetInterfaceNameFromGuid(GUID AdapterGUID, DWORD par1, DWORD par2)
+{
+    DWORD ApiReturn, Error;
+    ULONG ulOutBufLen = 0;
+    GUID UniqueGUID;
+    WCHAR Name[MAX_INTERFACE_NAME_LEN];
+
+    ulOutBufLen = 0;
+
+    // Test NULL GUID
+    SetLastError(0xbeeffeed);
+    ZeroMemory(&Name, sizeof(Name));
+    _SEH2_TRY
+    {
+        ApiReturn = pTestFunction(NULL, &Name, &ulOutBufLen, par1, par2);
+    }
+    _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+    {
+        Error = _SEH2_GetExceptionCode();
+        ok(Error == STATUS_SUCCESS, "Exception 0x%08lx, expected 0x%08lx\n", Error, STATUS_SUCCESS);
+    }
+    _SEH2_END;
+    Error = GetLastError();
+
+    ok(ApiReturn == ERROR_INSUFFICIENT_BUFFER, // Windows 7: ERROR_INVALID_PARAMETER
+       "ApiReturn returned %ld, expected ERROR_INSUFFICIENT_BUFFER\n",
+       ApiReturn);
+    ok(Error == 0xbeeffeed,
+       "GetLastError() returned %ld, expected 0xbeeffeed\n",
+       Error);
+    ok(ulOutBufLen == MAX_INTERFACE_NAME_LEN * 2, // Windows 7: 0
+       "ulOutBufLen is %ld, expected = MAX_INTERFACE_NAME_LEN * 2\n",
+       ulOutBufLen);
+    Error = wcslen(Name);
```

Comment:
Why not `ok_wstr(L"", Name)`?
That will print the name if it fails, instead of your 6 line piece of code that _sometimes_ traces the name.

Reply:
Didn't know about this useful function :)